contracts/v2/sovereignChains/BridgeL2SovereignChain.sol

```diff
@@ -440,20 +480,146 @@ contract BridgeL2SovereignChain is
         }
     }
 
+    /*
+     * @notice unset a claim from the claimedBitmap
+     * @param leafIndex Index
+     * @param sourceBridgeNetwork Origin network
+     */
+    function _unsetClaimedBitmap(
+        uint32 leafIndex,
+        uint32 sourceBridgeNetwork
+    ) private {
+        uint256 globalIndex = uint256(leafIndex) +
+            uint256(sourceBridgeNetwork) *
+            _MAX_LEAFS_PER_NETWORK;
+        (uint256 wordPos, uint256 bitPos) = _bitmapPositions(globalIndex);
+        uint256 mask = ~(1 << bitPos);
+        // Check if the bit is already unset
+        if ((claimedBitMap[wordPos] & (1 << bitPos)) == 0) {
+            revert ClaimNotSet();
+        }
+        // Use bitwise AND with the negated mask to unset the bit
+        claimedBitMap[wordPos] &= mask;
```

Comment:
i think the code that we have it's more optimal in terms of gas, and also it's more similar, so easier to audit, i woudl change it for:

  uint256 mask = 1 << bitPos;
        uint256 flipped = claimedBitMap[wordPos] ^= mask;
        if (flipped & mask != 0) {
            revert ClaimNotSet();
        }